src/irmin/tree.ml

```diff
@@ -1989,39 +1996,48 @@ module Make (P : Private.S) = struct
 
     let to_tree p =
       let env = Env.empty () in
-      Env.to_mode env Deserialise;
+      Env.to_mode env Env.Structural Env.Deserialise;
```

Comment:
This `to_mode` function is confusing. Should it be called `Env.set_mode` instead?